src/pages/register-a-beacon/beacon-information.tsx

```diff
@@ -2,65 +2,114 @@ import { GetServerSideProps } from "next";
 import React, { FunctionComponent } from "react";
 import { BackButton, Button } from "../../components/Button";
 import {
-  DateInput,
   DateListInput,
   DateListItem,
   DateType,
 } from "../../components/DateInput";
 import { Details } from "../../components/Details";
+import { FormErrorSummary } from "../../components/ErrorSummary";
 import {
   Form,
   FormFieldset,
   FormGroup,
-  FormHint,
-  FormLabel,
   FormLegendPageHeading,
 } from "../../components/Form";
 import { Grid } from "../../components/Grid";
-import { Input } from "../../components/Input";
+import { FormInputProps, Input } from "../../components/Input";
 import { InsetText } from "../../components/InsetText";
 import { Layout } from "../../components/Layout";
 import { IfYouNeedHelp } from "../../components/Mca";
 import { FieldManager } from "../../lib/form/fieldManager";
 import { FormManager } from "../../lib/form/formManager";
+import { Validators } from "../../lib/form/validators";
 import { CacheEntry } from "../../lib/formCache";
-import { handlePageRequest } from "../../lib/handlePageRequest";
+import { FormPageProps, handlePageRequest } from "../../lib/handlePageRequest";
+import { padNumberWithLeadingZeros } from "../../lib/utils";
+
+interface DateInputProps {
+  monthValue: string;
+  yearValue: string;
+  errorMessages: string[];
+}
+
+function getISODate(year: string, month: string): string {
+  if (year !== "" && month !== "") {
+    try {
+      return new Date(parseInt(year), parseInt(month) - 1).toISOString();
```

Comment:
See constructor for [JS Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#syntax)

Reply:
I don't know how likely this situation is, but was playing around with the date input and was wondering why the form didn't moan at me to enter only integers here:
![image](https://user-images.githubusercontent.com/32230328/110107868-5d191c00-7da3-11eb-8e5d-c61265184632.png)

And realised that parseInt is a little too forgiving maybe:
<img width="515" alt="image" src="https://user-images.githubusercontent.com/32230328/110107763-3ce95d00-7da3-11eb-9d1a-095cc975c991.png">

Maybe `Number()` might be better for here? https://stackoverflow.com/a/4564199

Reply:
Thank you @Ninamma! 🙏  Good spot, I have updated and reflected in the cypress tests as well